controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Maintainer.java

```diff
@@ -27,16 +30,23 @@
     private final JobControl jobControl;
     private final ScheduledExecutorService service;
     private final String name;
+    private List<SystemName> systemNames;
 
     public Maintainer(Controller controller, Duration interval, JobControl jobControl) {
         this(controller, interval, jobControl, null);
     }
 
+    public Maintainer(List<SystemName> systemNames, Controller controller, Duration interval, JobControl jobControl) {
```

Comment:
Consider keeping the number of overloaded constructors low, e.g. one simplified (like the above) and one that has all possible parameters.